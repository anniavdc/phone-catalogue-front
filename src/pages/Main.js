import React from 'react';

// libraries
import {
    Route,
    Switch,
    Redirect,
    useHistory
} from 'react-router-dom';

// components
import PhoneListPage from './phones/PhoneListPage';
import PhoneDetail from './phones/PhoneDetail';

export default function Main({ match }) {
    let history = useHistory();

    return (
        <div className="main-page">
            <header className="main-header">
                <h1 className="main-title" onClick={() => history.push(match.path)}>Product app</h1>
            </header>
            <section className="main-content">
                <Switch>
                    <Route exact path={`${match.path}/:phoneId`} component={PhoneDetail} />
                    <Route exact path={match.path} component={PhoneListPage} />
                    <Redirect to={match.path} />
                </Switch>
            </section>
            <footer className="main-footer">
                by Annia da Costa Martins
            </footer>
        </div>
    )
}
