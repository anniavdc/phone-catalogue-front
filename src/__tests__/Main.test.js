import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { fireEvent } from '@testing-library/dom';

// files
import render from '../utils/tests/renderWithRouter';
import Main from '../pages/Main';

const initialState = {
    phones: {
        phoneList: [
            {
                color: "black",
                description: "This is iPhone 7.",
                id: 0,
                imageFileName: "IPhone_7.png",
                manufacturer: "Apple",
                name: "iPhone 7",
                price: 769,
                processor: "A10 Fusion",
                ram: 2,
                screen: "4,7 inch IPS"
            }
        ]
    }
};

const mockStore = configureStore([thunk]);

describe('Main', () => {
    let store;
    beforeEach(() => {
        store = mockStore(initialState);
    })

    test('render', () => {
        const { container } = render(
            <Provider store={store}>
                <Main match={{ params: {}, isExact: true, path: "/app", url: "/" }} />
            </Provider>

        );
        expect(container).toMatchSnapshot();
    });

    test('should redirect to app', async () => {
        const { history, container } = render(
            <Provider store={store}>
                <Main match={{ params: { phoneId: "0" }, isExact: true, path: "/app/:phoneId", url: "app/0" }} />
            </Provider>,
            "/app/0",
            { wrapper: MemoryRouter }
        );

        const homeLink =  container.querySelector(".main-title");
        fireEvent.click(homeLink)

        await expect(history.location.pathname).toMatch(/app/);

    });

})
