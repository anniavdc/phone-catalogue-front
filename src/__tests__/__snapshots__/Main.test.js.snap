// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render 1`] = `
<div>
  <div
    class="main-page"
  >
    <header
      class="main-header"
    >
      <h1
        class="main-title"
      >
        Product app
      </h1>
    </header>
    <section
      class="main-content"
    >
      <h2
        class="products-title"
      >
        Phone Catalogue
      </h2>
      <div
        class="products-wrapper"
      >
        <div
          class="product-card"
        >
          <p
            class="card-info"
          >
            769â‚¬
          </p>
          <div
            class="card-image"
            style="background-image: url(http://localhost:5000/public/IPhone_7.png);"
          />
          <div>
            <h3
              class="card-title"
            >
              iPhone 7
            </h3>
            <p
              class="card-content"
            >
              Apple
            </p>
          </div>
        </div>
      </div>
    </section>
    <footer
      class="main-footer"
    >
      by Annia da Costa Martins
    </footer>
  </div>
</div>
`;
